src/test/lib/logical_query_plan/join_node_test.cpp

```diff
@@ -678,6 +688,177 @@ TEST_P(JoinNodeMultiJoinModeTest, UniqueColumnCombinationsMultiPredicateJoin) {
+    EXPECT_TRUE(order_dependencies.contains(od_v_to_u));
+  }
+
+  // Case (iii): Multiple join predicates.
+  {
+    // clang-format off
+    const auto inner_join_node =
+    JoinNode::make(JoinMode::Inner, expression_vector(equals_(_t_a_a, _t_c_u), equals_(_t_a_b, _t_c_v)),
+      _mock_node_a,
+      _mock_node_c);
+    // clang-format on
+    const auto& order_dependencies = inner_join_node->order_dependencies();
+    EXPECT_EQ(order_dependencies.size(), 2);
+    EXPECT_TRUE(order_dependencies.contains(od_a_to_c));
+    EXPECT_TRUE(order_dependencies.contains(od_v_to_u));
+  }
+
+  // Case (iv): OD from left input leads to new ODs with columns from the right input.
+  {
+    const auto od_b_to_a = OrderDependency{{_t_a_b}, {_t_a_a}};
+    const auto od_y_to_x = OrderDependency{{_t_b_y}, {_t_b_x}};
+    const auto order_constraint = TableOrderConstraint{{ColumnID{1}}, {ColumnID{0}}};
+    _mock_node_a->set_order_constraints({order_constraint});
+    _mock_node_b->set_order_constraints({order_constraint});
+
+    const auto& order_dependencies = _inner_join_node->order_dependencies();
+    EXPECT_EQ(order_dependencies.size(), 7);
+    EXPECT_TRUE(order_dependencies.contains(od_b_to_a));
+    EXPECT_TRUE(order_dependencies.contains(od_y_to_x));
+    EXPECT_TRUE(order_dependencies.contains(OrderDependency{{_t_a_a}, {_t_b_y}}));
+    EXPECT_TRUE(order_dependencies.contains(OrderDependency{{_t_b_y}, {_t_a_a}}));
+
+    EXPECT_TRUE(order_dependencies.contains(OrderDependency{{_t_a_b}, {_t_b_y}}));
+    EXPECT_TRUE(order_dependencies.contains(OrderDependency{{_t_a_b}, {_t_b_x}}));
+    EXPECT_TRUE(order_dependencies.contains(OrderDependency{{_t_a_a}, {_t_b_x}}));
+  }
+
+  // Case (v): OD from right input leads to new ODs with columns from the left input.
+  {
+    const auto od_a_to_b = OrderDependency{{_t_a_a}, {_t_a_b}};
+    const auto od_x_to_y = OrderDependency{{_t_b_x}, {_t_b_y}};
+    const auto order_constraint = TableOrderConstraint{{ColumnID{0}}, {ColumnID{1}}};
+    _mock_node_a->set_order_constraints({order_constraint});
+    _mock_node_b->set_order_constraints({order_constraint});
+
+    const auto& order_dependencies = _inner_join_node->order_dependencies();
+    EXPECT_EQ(order_dependencies.size(), 7);
+    EXPECT_TRUE(order_dependencies.contains(od_a_to_b));
+    EXPECT_TRUE(order_dependencies.contains(od_x_to_y));
+    EXPECT_TRUE(order_dependencies.contains(OrderDependency{{_t_a_a}, {_t_b_y}}));
+    EXPECT_TRUE(order_dependencies.contains(OrderDependency{{_t_b_y}, {_t_a_a}}));
+
+    EXPECT_TRUE(order_dependencies.contains(OrderDependency{{_t_b_x}, {_t_a_a}}));
+    EXPECT_TRUE(order_dependencies.contains(OrderDependency{{_t_b_x}, {_t_a_b}}));
+    EXPECT_TRUE(order_dependencies.contains(OrderDependency{{_t_b_y}, {_t_a_b}}));
+  }
+
+  // Case (vi): Self join. No new ODs should be built. Inner equi join on a = c and ODs [a] |-> [b] and [a] |-> [c]
+  // would lead to [c] |-> [a] and [c] |-> [b] normally.
```

Comment:
```c++
// Case (vi): Self join. No new ODs should be built. The inner equi join on a = c with input ODs [a] |-> [b] and
//            [a] |-> [c] would lead to [c] |-> [a] and [c] |-> [b], normally. However, we cannot distinguish WHICH
//            version of c orders b (only the one that equals a in the output).
```